data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/utils/LogPusher.java

```diff
@@ -0,0 +1,88 @@
+package org.opensearch.dataprepper.plugins.sink.utils;
+
+import io.micrometer.core.instrument.Counter;
+import org.opensearch.dataprepper.plugins.sink.buffer.Buffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.CloudWatchLogsException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InputLogEvent;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsRequest;
+
+import java.util.ArrayList;
+
+public class LogPusher {
```

Comment:
Please add some unit tests for this class.

Reply:
Thank you, I will have the unit tests for this class ready for the next commit.